supersetapiclient/databases.py

```diff
@@ -1,36 +1,70 @@
 """Databases."""
 from dataclasses import dataclass
+from typing import Optional
 
 from supersetapiclient.base import (
-    Object, ObjectFactories, json_field, default_string
+    Object, ObjectFactories, default_string, json_field
 )
 
 
 @dataclass
 class Database(Object):
-    JSON_FIELDS = ["metadata_cache_timeout"]
+    JSON_FIELDS = [
+        "extra",
+    ]
 
-    id: int
     database_name: str
-    allow_ctas: bool
-    allow_cvas: bool
-    allow_dml: bool
-    allow_multi_schema_metadata_fetch: bool
-    allow_run_async: bool
-    metadata_cache_timeout: dict = json_field()
+    id: Optional[int] = None
+    allow_ctas: bool = True
+    allow_cvas: bool = True
+    allow_dml: bool = True
+    allow_multi_schema_metadata_fetch: bool = True
+    allow_run_async: bool = True
+    expose_in_sqllab: bool = True
+    cache_timeout: Optional[int] = None
     encrypted_extra: str = default_string()
     engine: str = default_string()
-    extra: str = json_field()
+    extra: dict = json_field()
     force_ctas_schema: str = default_string()
     server_cert: str = default_string()
     sqlalchemy_uri: str = default_string()
 
+    def to_json(self, *args, **kwargs):
```

Comment:
I see that you add to_json in many class. Do you have a specific use case in mind ?

Reply:
`to_json()` is used in `some_entity.add()` and `MyItem.save()` to convert from Python object instance to its JSON representation.